### DISCUSSION TITLE
Image onError callback not getting called

### DISCUSSION THREAD

**Discussion by ahetawal-p** [mantinedev/mantine#3640] [Q&A]
I have a basic image component in a NextJS app, and due to server data, there might be cases when the actual image src is invalid or unreachable. 
I am trying to listen to the onError callback to set the source to a different image in that case.
Here is my basic setup.

```
const [imageUrl, setImageUrl] = useState<string>(originalSrc)

const handleError = (e: unknown) => {
   console.log('see me')
   setImageUrl(placeHolders[0])
  }

  return (
    <Image
      src={imageUrl}
      alt={alt ?? ''}
      onError={handleError} // doesnt work
      // imageProps={{ onError: handleError }} doesnt work
    />
  )
```

I am not able to get the onError callbacks.
How can I achieve this with Image component. 
The nextjs image component seems to be working fine with the similar setup.

**Comment by rtivital** [comment]
If you use Next.js, it is probably a good idea to use next/image instead of Mantine Image component

**Reply by ahetawal-p** [reply]
Sure, but it comes with its own requirements. We were just looking for a light weight solution as we are fully mantine based app.

**Reply by rtivital** [reply]
You are welcome to explore the source code and send a PR with a fix
